pkg/cloudprovider/gce/gce.go

```diff
@@ -480,11 +482,14 @@ func (gce *GCECloud) NodeAddresses(_ string) ([]api.NodeAddress, error) {
 	if err != nil {
 		return nil, fmt.Errorf("couldn't get external IP: %v", err)
 	}
+	internalIP, err := gce.metadataAccess(INTERNAL_IP_METADATA_URL)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't get external IP: %v", err)
+	}
 
 	return []api.NodeAddress{
 		{Type: api.NodeExternalIP, Address: externalIP},
-		// TODO(mbforbes): Remove NodeLegacyHostIP once v1beta1 is removed.
-		{Type: api.NodeLegacyHostIP, Address: externalIP},
+		{Type: api.NodeInternalIP, Address: internalIP},
```

Comment:
For clarity, I'd put these in the priority order expected by GetNodeHostIP: InternalIP should go first.